exec/java-exec/src/main/codegen/templates/MapWriters.java

```diff
@@ -110,9 +110,9 @@ public ListWriter list(String name) {
     FieldWriter writer = fields.get(name.toLowerCase());
     if(writer == null) {
       writer = new SingleListWriter(name, container, this);
-      writer.setPosition(${index});
       fields.put(name.toLowerCase(), writer);
     }
+    writer.setPosition(${index});
```

Comment:
Your change seems to violate the design pattern of the writers. Position is managed via the setPosition call at line 152. If you're seeing problems, you should figure who isn't calling that method. I believe the code before your change is correct: the only reason we should have to reposition a writer when we retrieve it is it didn't previously exist and we had to create a new one. This is the pattern used across the board for position in the context of all the writers. Position is managed externally and only a new field creation should cause a location position set.